commands/operator-sdk/cmd/add/api.go

```diff
@@ -162,7 +165,54 @@ func apiRun(cmd *cobra.Command, args []string) {
 		log.Fatalf("failed to create %v: %v", filePath, err)
 	}
 
-	// TODO: append rbac rule to deploy/rbac/role.yaml
+	// append rbac rule to deploy/role.yaml
+	roleFilePath := filepath.Join(fullProjectPath, "deploy", "role.yaml")
+	roleYAML, err := ioutil.ReadFile(roleFilePath)
+	if err != nil {
+		log.Fatalf("failed to read role manifest %v: %v", roleFilePath, err)
+	}
+	obj, _, err := cgoscheme.Codecs.UniversalDeserializer().Decode(roleYAML, nil, nil)
+	if err != nil {
+		log.Fatalf("failed to decode role manifest %v: %v", roleFilePath, err)
+	}
+	switch role := obj.(type) {
+	// TODO: use rbac/v1.
+	case *rbacv1beta1.Role:
+		pr := &rbacv1beta1.PolicyRule{}
+		apiGroupFound := false
+		for i := range role.Rules {
+			if role.Rules[i].APIGroups[0] == r.FullGroup {
+				apiGroupFound = true
+				pr = &role.Rules[i]
+				break
+			}
+		}
+		// check if the resource already exists
+		for _, resource := range pr.Resources {
+			if resource == r.Resource {
+				log.Fatalf("apiGroup (%v) resource (%v) already exists", pr.APIGroups[0], resource)
+			}
+		}
+
+		pr.Resources = append(pr.Resources, r.Resource)
+		// create a new apiGroup if not found.
+		if !apiGroupFound {
+			pr.APIGroups = []string{r.FullGroup}
+			pr.Resources = []string{r.Resource}
+			pr.Verbs = []string{"*"}
+			role.Rules = append(role.Rules, *pr)
+		}
+		// update role.yaml
+		data, err := yaml.Marshal(&role)
+		if err != nil {
+			log.Fatalf("failed to marshal role(%+v): %v", role, err)
+		}
+		if err := ioutil.WriteFile(roleFilePath, data, cmdutil.DefaultFileMode); err != nil {
+			log.Fatalf("failed to update %v: %v", roleFilePath, err)
+		}
+	default:
+		log.Fatal("failed to parse role.yaml as a role")
+	}
```

Comment:
Can you group this into a helper function below so the `add api` command stays a bit more readable:
```Go
if err := updateRoleForResource(r); err != nil {
    log.Fatal("failed to update RBAC manifest for resource (%v/%v): %v", r.APIVersion, r.Kind, err)
}

func updateRoleForResource(r *scaffold.Resource) error{
    roleFilePath := filepath.Join(fullProjectPath, "deploy", "role.yaml")
...
}
```

Reply:
yeah, sure thing.